File: test-backend.js
const https = require('https');

// Test health endpoint
console.log('Testing health endpoint...');
const healthReq = https.get('http://lana-ai.onrender.com/health', (res) => {
  console.log(`Health status code: ${res.statusCode}`);
  
  let data = '';
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    console.log('Health response:', data);
    
    // Test structured lesson endpoint
    console.log('\nTesting structured lesson endpoint...');
    const postData = JSON.stringify({
      topic: 'Mahogany',
      age: 10
    });
    
    const options = {
      hostname: 'lana-ai.onrender.com',
      port: 80,
      path: '/api/structured-lesson',
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(postData)
      }
    };
    
    const lessonReq = https.request(options, (res) => {
      console.log(`Lesson status code: ${res.statusCode}`);
      
      let lessonData = '';
      res.on('data', (chunk) => {
        lessonData += chunk;
      });
      
      res.on('end', () => {
        console.log('Lesson response:', lessonData);
        
        try {
          const jsonData = JSON.parse(lessonData);
          if (jsonData.introduction && jsonData.introduction.includes('Unable to generate a detailed lesson')) {
            console.log('✅ SUCCESS: Error response detected instead of hardcoded template');
          } else if (jsonData.introduction && jsonData.introduction.startsWith('Let\'s learn about')) {
            console.log('❌ FAILURE: Still receiving hardcoded template response');
          } else {
            console.log('✅ SUCCESS: Received dynamic content (likely from LLM)');
          }
        } catch (e) {
          console.log('Response is not valid JSON');
        }
      });
    });
    
    lessonReq.on('error', (e) => {
      console.error('Lesson request error:', e.message);
    });
    
    lessonReq.write(postData);
    lessonReq.end();
  });
});

healthReq.on('error', (e) => {
  console.error('Health request error:', e.message);
});

healthReq.end();

File: test-backend.py
import requests
import json

# Test health endpoint
print("Testing health endpoint...")
try:
    health_response = requests.get("http://lana-ai.onrender.com/health")
    print(f"Health status code: {health_response.status_code}")
    print(f"Health response: {health_response.text}")
    
    # Test structured lesson endpoint
    print("\nTesting structured lesson endpoint...")
    lesson_data = {
        "topic": "Mahogany",
        "age": 10
    }
    
    lesson_response = requests.post(
        "http://lana-ai.onrender.com/api/structured-lesson",
        headers={"Content-Type": "application/json"},
        data=json.dumps(lesson_data)
    )
    
    print(f"Lesson status code: {lesson_response.status_code}")
    print(f"Lesson response: {lesson_response.text}")
    
    if lesson_response.status_code == 200:
        try:
            json_data = lesson_response.json()
            if "introduction" in json_data and "Unable to generate a detailed lesson" in json_data["introduction"]:
                print("✅ SUCCESS: Error response detected instead of hardcoded template")
            elif "introduction" in json_data and json_data["introduction"].startswith("Let's learn about"):
                print("❌ FAILURE: Still receiving hardcoded template response")
            else:
                print("✅ SUCCESS: Received dynamic content (likely from LLM)")
                print(f"Introduction preview: {json_data.get('introduction', '')[:100]}...")
        except json.JSONDecodeError:
            print("Response is not valid JSON")
    else:
        print(f"Request failed with status code {lesson_response.status_code}")
        
except requests.exceptions.RequestException as e:
    print(f"Request error: {e}")